### PR TITLE
[#1595] Add release branch usage and hot patching instructions

### LINKED ISSUE #1595: Dev Guide: Tweak documentation of doing a release
Suggested tweaks to documentation:
* Before doing a release, merge master to the release branch, and do the release using the release branch rather than the master branch.
* Use a merge commit for the above, so that we have a visual history of releases 

We can also document the procedure for hot-patching a past release
1. PR against the release branch
2. Merge the PR, and release a new version (using the release branch)
3. Merge the release branch back to the master branch

The above is what we do in some of the other NUS-OSS projects, but I'm open to other suggestions.

### PR DESCRIPTION
**Pull request by HCY123902**
Fixes #1595.

```
There is currently no specific instruction on how to update the
release branch before making a release, or how to handle critical bugs
spotted after the release is made.

Adding these instructions will reduce the communication expenses in
the project release.

Let's adjust the release guide accordingly.
```

### CODE REVIEW

**Comment by HCY123902** [comment]
@dcshzj Can you help to review this pull request? Many Thanks

File: docs/dg/projectManagement.md [hunk 1]
```diff
@@ -37,10 +41,11 @@ Take note of the following when making the release according to the above guide:
   * Append `rc` to the version number to indicate that the release is a pre-release that is not ready to be used in production.
 * Enter the release title as `RepoSense vxxx` where `xxx` is the version number. Enter the release description by referring to the previous [RepoSense releases](https://github.com/reposense/RepoSense/releases).
 * Before launching the release, generate the `RepoSense.jar` file and attach it to the release.
-  * Change the directory to the project root directory.
-  * In the terminal, run `gradlew --version` to check that the JDK version is 1.8.0.
-  * Run `gradlew shadowJar`, and the Jar file will be generated at `{buildDir}/jar/`.
-  * Check that the Jar file is working. You may need to check that the report can be generated from the Jar file both locally and remotely by following the [Generating Reports Guide](../ug/generatingReports.html).
+  1. Switch to the `release` branch.
+  1. In the terminal, Change the directory to the project root directory.
```
Comment (dcshzj) [inline docs/dg/projectManagement.md:45]:
```suggestion
  1. In the terminal, change the directory to the project root directory.
```

Lower-case after comma

**Review by dcshzj — CHANGES_REQUESTED** [review]

File: docs/dg/projectManagement.md (hunk 1 above)
Comment (HCY123902) [reply docs/dg/projectManagement.md]: Thanks for noticing this. The adjustment has been made

**Comment by HCY123902** [comment]
@dcshzj Hello, I have made the corresponding adjustment. Can you take a look again

**Review by dcshzj — APPROVED** [review]
LGTM!

**Pull request merged**